collection-views.js

```diff
@@ -90,13 +117,56 @@ CollectionView.prototype._mutateSelector = function (selector, query) {
       // If narrowingQuery is a function we should pass the result of calling that function
       if (_.isFunction(narrowingQuery))
         narrowingQuery = narrowingQuery();
-      _.extend(query, narrowingQuery)
+      _.extend(query, narrowingQuery);
     }
-    collection = collection._parentCollection
+    collection = collection._parentCollection;
   }
 
   if (LocalCollection._selectorIsId(selector))
     selector = {_id: selector};
 
   return _.extend({}, selector, query);
+};
+
+/**
+ * @summary This function builds the options object by extending the parent's options object
+ * @locus Anywhere
+ * @protected
+ * @method _mutateOptions
+ * @memberOf CollectionView
+ * @param {Object | String} The input options
+ * @return {Object} The chained options object
+ */
+CollectionView.prototype._mutateOptions = function (options) {
+  var collection = this
+    , options = options || {};
+
+  while (collection) {
+    if (! _.isUndefined(collection._narrowingOptions)) {
+      _.each(collection._narrowingOptions, function (val, key) {
+        if (! _.isUndefined(options[key]))
+          _.extend(options[key], val);
```

Comment:
This won't always restrict the set of fields which are returned. Sometimes it will expand the set of fields which are returned. For example `Books.where({}, {genre: 1, title: 1}).where({}, {genre: 1})` should not return the title of books.